src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -636,7 +670,13 @@ SHARED_API int __cdecl corehost_initialize(const corehost_initialize_request_t *
 
     if (rc == StatusCode::CoreHostAlreadyInitialized)
     {
-        // [TODO] Compare the current context with this request (properties)
+        assert(init_request != nullptr
+            && init_request->version >= offsetof(corehost_initialize_request_t, config_values) + sizeof(init_request->config_values)
+            && init_request->config_keys.len == init_request->config_values.len);
+
+        // Compare the current context with this request (properties)
+        if (!matches_existing_properties(init_request))
+            rc = StatusCode::CoreHostDifferentProperties;
```

Comment:
This seems strange.  I thought we were going to let the second client observe the properties and make the decision for themselves. This seems to be forcing failure.

Reply:
`StatusCode::CoreHostDifferentProperties` is a "warning" error code. It will be documented that it's perfectly valid to use the returned context.

Reply:
It is intended to still be considered success and not failure (just like `StatusCode::CoreHostAlreadyInitialized`). I basically stole the COM approach and made > 0 values different success codes.

Would it be clearer to give the success codes some prefix (`Success*`? `Warning*`?) indicating they are not actually failures?

Reply:
That's a good idea...
`Success_DifferentRuntimeProperties` - would make lot of sense to me.